streams/src/test/java/org/apache/kafka/streams/processor/internals/RecordCollectorTest.java

```diff
@@ -570,7 +570,7 @@ public Optional<Set<Integer>> partitions(final String topic, final String key, f
     public void shouldUseDefaultPartitionerAsPartitionReturnsNull() {
 
         final StreamPartitioner<String, Object> streamPartitioner =
-                (topic, key, value, numPartitions) -> null;
+                (topic, key, value, numPartitions) -> Optional.empty();
```

Comment:
The test name should be updated... we used `null` for the old method, but `empty` for the new one, which we test now.